### DISCUSSION TITLE
Feature Request: Ability to bypass status checks/mergability reqs to enter the merge queue

### DISCUSSION THREAD

**Discussion by Ryang20718** [trunk-io/docs#372] [Feature Requests]
Use cases:

- We'd like to make our code cov status checks merge blocking, but certain PR's may certainly have 0% patch coverage (only touches main that isn't tested)

- We'd like the ability to bypass status checks on reverts without forcing the PR in directly

ideally something like

`/trunk merge <pr> bypass`

**Comment by pv72895** [comment]
Hey Ryan!

So right now, the way that merge works is that we wait until GitHub tells us that the PR is mergeable before letting it into the queue. This is because we want to make sure we won’t get blocked and have to restart the whole queue if we try to merge it and GitHub blocks us

Phrased differently - I think the proper case here is that if you want to make a code coverage check blocking, but not when it’s 0%, you have to make that check report a success, or else github itself will block you

Otherwise, if it’s reported as a failure and is a required status in GitHub’s eyes, then when we try to merge it, GitHub will block us. If GitHub allowed you to make it so the Trunk GitHub app could bypass individual checks, then that would work, but I don't think that's possible
